src/openrct2/rct2/S6Exporter.cpp

```diff
@@ -496,8 +496,27 @@ void S6Exporter::ExportRide(rct2_ride* dst, const Ride* src)
 
     // pad_046;
     dst->status = src->status;
-    dst->name = src->name;
-    dst->name_arguments = src->name_arguments;
+    if (src->custom_name.empty())
+    {
+        // Default name with number
+        dst->name = RideNaming[src->type].name;
```

Comment:
This will (among other things) cause names like "Shop 1", "Shop 2", etc in exported saves.

Reply:
While this is true, using allocated string ids (which is required to get the object name) relies on vanilla allocating the same string IDs to those object names right?

Reply:
Yes, but so far that _has_ worked correctly.

Reply:
I have just done a test. It works correctly if using a name from RideNaming or from a rideEntry, but if it is named after a ride group, it error trappers.

Reply:
Yeah, well we can basically deprecate compatibility with RCT2 vanilla at this point. We will soon be removing it altogether.